server/src/routes/api/v1/voteRouter.js

```diff
@@ -0,0 +1,44 @@
+import express from "express";
+import { Review, Vote } from "../../../models/index.js";
+import { ValidationError } from "objection";
+import ReviewSerializer from "../../../serializers/ReviewSerializer.js"
+
+const voteRouter = new express.Router();
+
+voteRouter.post("/", async (req, res) => {
+    const voteValue = req.body.voteValue;
+    const reviewId = req.body.reviewId;
+    const userId = req.user.id;
+    try {
+        const voteExists = await Vote.query().where("reviewId",`${reviewId}`).where("userId",`${userId}`)
+        if (voteExists.length>0) {
+            if(voteValue === voteExists[0].voteValue) {
+                await Vote.query().delete().where("reviewId",`${reviewId}`).where("userId",`${userId}`)
+            } else { 
+                await Vote.query().delete().where("reviewId",`${reviewId}`).where("userId",`${userId}`)
+                const insertOppositeVote = await Vote.query().insertAndFetch({
+                    voteValue,
+                    userId,
+                    reviewId
+                });
+            }
+        } else {
+            const newVote = await Vote.query().insertAndFetch({
+                voteValue,
+                userId,
+                reviewId
+            });
+        }
+        const review = await Review.query().findById(reviewId)
+        const serializedReview = await ReviewSerializer.singleShowDetails(review)
+        return res.status(201).json({ review: serializedReview });
```

Comment:
you send up the entire Review object as JSON here, but we only need the new vote total for this review. We can save on queries by replacing line 32 and 33 with 
- a `where` query to grab all votes that have a `reviewId` column of the `reviewId` variable you have
- a count of those returned votes

essentially, move the vote query logic you have in `singleShowDetails` into here, and ignore getting the review from the database at all.